core/context.go

```diff
@@ -22,6 +24,10 @@ type Context struct {
 	// Frame receives from client.
 	Frame frame.Frame
 
+	// Metadata is be built from handshake.
```

Comment:
Add `Metadata` and `Route` to the Context object, since they are built in the handshake stage and can be used after the handshake is complete.